internal/runtime/internal/worker/worker_pool.go

```diff
@@ -69,9 +72,20 @@ func (w *fixedWorkerPool) QueueSize() int {
 	return w.workQueue.queueSize()
 }
 
-func (w *fixedWorkerPool) Stop() {
+func (w *fixedWorkerPool) Stop(timeout time.Duration) error {
 	close(w.quit)
-	w.allStopped.Wait()
+	done := make(chan struct{})
+	go func() {
```

Comment:
Isn't this goroutine going to keep running even after `case <-time.After(timeout)`?